integration_tests/run_pyspark_from_build.sh

```diff
@@ -117,10 +117,15 @@ else
     NUM_LOCAL_EXECS=${NUM_LOCAL_EXECS:-0}
     MB_PER_EXEC=${MB_PER_EXEC:-1024}
     CORES_PER_EXEC=${CORES_PER_EXEC:-1}
-
-    if ((NUM_LOCAL_EXECS > 0)); then
-      export PYSP_TEST_spark_master="local-cluster[$NUM_LOCAL_EXECS,$CORES_PER_EXEC,$MB_PER_EXEC]"
-    fi
+    # Spark 3.1.1 includes https://github.com/apache/spark/pull/31540
+    # which helps with spurious task failures as observed in our tests. If you are running
+    # Spark versions before 3.1.1, this sets the spark.max.taskFailures to 4 to allow for
+    # more lineant configuration, else it will set them to 1 as spurious task failures are not expected
+    # for Spark 3.1.1+
+    VERSION_STRING=`unzip -q -c $SPARK_HOME/jars/spark-core_*.jar spark-version-info.properties | head -1 | cut -d'=' -f2`
```

Comment:
Is it guaranteed that the Spark version is the first thing in the properties file across all possible Spark versions (e.g.: from cloud providers, etc.)?  I think this should be filtering for the `version` property (e.g.: replace `head -1` with `grep '^version='`) and erroring when it cannot find it.

Reply:
It is according to: https://github.com/apache/spark/blob/master/build/spark-build-info#L29. But I can make it stronger with the grep as suggested + error if the version can't be detected this way.

Reply:
Yes, it's that way in that particular codebase, but not necessarily in a future Spark version or in some cloud provider's custom version.  Checking for the well-known property name seems prudent here and lower risk than assuming what is in the first line of a property file that is otherwise semantically unordered.

Reply:
Please take a look at https://github.com/NVIDIA/spark-rapids/pull/2792/commits/0fdea2c0ff2169172659c6dae2add2e3fe7420a8, which should handle this case.